src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -845,13 +845,21 @@ namespace shogun
 		 * represented
 		 * as \f$\sum_i a_i b_i\f$
 		 */
-		template <typename T, typename U, typename TU = typename promote<T, U>::type>
-		TU dot(const SGVector<T>& a, const SGVector<U>& b)
+		template <
+		    typename T, typename U, typename TU = typename promote<T, U>::type,
+		    typename Tag = void*>
```

Comment:
what i mean is you can have here instead `template<..., bool allow_cast = false>` and that is compile time evaluated no?

Reply:
Ah yes i misunderstood you. But what about type deduction? We would have to specify T, U and TU before we get to the boolean.

Reply:
Ah ok, could you put the Boolean first?

Reply:
That would work :)

Reply:
It's just a suggestion btw!

Reply:
I find tags more expressive, so I will use the name allow_cast, since likely_slow is horrible

Reply:
sounds good!